cmd/zfs/zfs_main.c

```diff
@@ -7500,59 +7501,82 @@ zfs_do_bookmark(int argc, char **argv)
 
 	/* check number of arguments */
 	if (argc < 1) {
-		(void) fprintf(stderr, gettext("missing snapshot argument\n"));
+		(void) fprintf(stderr, gettext("missing target argument\n"));
```

Comment:
I chose the `new` and `target` naming because the nvlist that is passed to the kernel associates `new => existing`.
But sure, `source` is less confusing because most users will think about it the other way around (I did too before learning about the nvlist structure).

How do we deal with the man page then, since the `source` (new term) now cannot only be a `snapshot` but also a `bookmark`? Also see https://github.com/zfsonlinux/zfs/pull/9571#discussion_r344594336 @Ornias1993

Reply:
Will go with `source`